src/test/java/teammates/test/cases/action/AdminActivityLogPageActionTest.java

```diff
@@ -1,27 +1,551 @@
+            int filteredLogs, Date earliestDateInUtc) {
+        AdminActivityLogPageAction action = getAction(params);
+        AjaxResult result = getAjaxResult(action);
+        AdminActivityLogPageData pageData = (AdminActivityLogPageData) result.data;
+        verifyStatusMessage(result.getStatusMessage(), totalLogs, filteredLogs, earliestDateInUtc);
+        verifyLogs(expected, pageData.getLogs());
+    }
+
+    private void verifyActionResult(int[][] expectedLogs, String... params) {
+        AdminActivityLogPageAction action = getAction(params);
+        ShowPageResult result = getShowPageResult(action);
+        AdminActivityLogPageData pageData = (AdminActivityLogPageData) result.data;
+        List<ActivityLogEntry> actualLogs = pageData.getLogs();
+        verifyLogs(expectedLogs, actualLogs);
+    }
+
+    /**
+     * Verifies actualLogs contains expectedLogs.
+     *
+     * <p>expectedLogs is a 2D array, the outer indices correspond to {@link #LOG_MESSAGE_INDEX_TODAY}
+     * {@link #LOG_MESSAGE_YESTDAY_INDEX} and {@link #LOG_MESSAGE_INDEX_TWO_DAYS_AGO}, the inner indices for
+     * every {@code LOG_MESSAGE_*_INDEX} correspond to the orders in the test data.
+     *
+     * @param expectedLogs
+     * @param actualLogs
+     */
+    private void verifyLogs(int[][] expectedLogs, List<ActivityLogEntry> actualLogs) {
+        List<String> expectedMsgs = generateExpectedMsgFrom(expectedLogs);
+
+        assertEquals(expectedMsgs.size(), actualLogs.size());
+        for (int i = 0; i < expectedMsgs.size(); i++) {
+            String actualMsg = actualLogs.get(i).generateLogMessage();
+            actualMsg = actualMsg.replace("<mark>", "").replace("</mark>", "");
+            assertTrue("expecte: " + expectedMsgs.get(i) + "to contain:" + actualMsg,
+                       expectedMsgs.get(i).contains(actualMsg));
+        }
+    }
+
+    private List<String> generateExpectedMsgFrom(int[][] expectedLogs) {
+        List<String> result = new ArrayList<String>();
+        for (int i = 0; i < expectedLogs.length; i++) {
+            for (int j = 0; j < expectedLogs[i].length; j++) {
+                result.add(logMessages.get(i).get(expectedLogs[i][j]));
+            }
+        }
+        return result;
+    }
+
+    private void verifyStatusMessage(String message, int totalLogs, int filteredLogs, Date earliestDateInUtc) {
+        SimpleDateFormat sdf = new SimpleDateFormat("dd-MM-yyyy HH");
+        sdf.setTimeZone(TimeZone.getTimeZone(Const.SystemParams.ADMIN_TIME_ZONE));
+
+        assertTrue(message.contains("Total Logs gone through in last search: " + totalLogs));
+        assertTrue(message.contains("Total Relevant Logs found in last search: " + filteredLogs));
+        assertTrue(message.contains("Logs are from following version(s): 1"));
+        assertTrue(message.contains("All available version(s): 1"));
+        // bug might be introduced when the time is 00:00 AM.
```

Comment:
Not sure how this comment is supposed to help the reader. May be provide some more info?